src/member/api.ts

```diff
@@ -97,37 +98,50 @@ export const getMemberStorageFiles = async (
     >(`${API_HOST}/${buildGetMemberStorageFilesRoute(pagination)}`)
     .then(({ data }) => data);
 
-export const editMember = async (
-  payload: {
-    id: UUID;
-    extra?: CompleteMember['extra'];
+export const editCurrentMember = async (
+  {
+    name,
+    extra,
+    enableSaveActions,
+  }: {
     name?: string;
+    extra?: CompleteMember['extra'];
     enableSaveActions?: boolean;
   },
   { API_HOST, axios }: PartialQueryConfigForApi,
 ) =>
-  verifyAuthentication(() =>
-    axios
-      .patch<CompleteMember>(
-        `${API_HOST}/${buildPatchMemberRoute(payload.id)}`,
-        {
-          extra: payload.extra,
-          name: payload.name?.trim(),
-          enableSaveActions: payload.enableSaveActions,
-        },
-      )
-      .then(({ data }) => data),
-  );
+  verifyAuthentication(() => {
+    const url = new URL(buildPatchCurrentMemberRoute(), API_HOST);
+    const body: Partial<
+      Pick<CompleteMember, 'extra' | 'name' | 'enableSaveActions'>
+    > = {};
+    if (name && name.trim() !== '') {
+      // trim name
+      body.name = name.trim();
+    }
+    if (extra && Object.keys(extra).length) {
+      body.extra = extra;
+    }
+    if (enableSaveActions !== undefined) {
+      body.enableSaveActions = enableSaveActions;
+    }
+    if (Object.keys(body).length) {
+      return axios
+        .patch<CompleteMember>(url.toString(), body)
+        .then(({ data }) => data);
+    } else {
+      return null;
+    }
```

Comment:
It's more a question, but I'm wondering if these checks are necessary here? Aren't we already checking in the frontend and in the backend? Also, what is happening in the frontend when returning null?

What do you think @pyphilia?

Reply:
I think it's okay to do those checks here. Maybe we want sth global so we indeed don't do the same checks manually everywhere? For now I would let it as done.
However you should maybe throw if the body is empty instead of returning null.